### PR TITLE
[feature](multi-catalog) parquet reader support nested array column

### PR DESCRIPTION
**Pull request by AshinGau**
# Proposed changes

Support to decode nested array column in parquet reader:
1. FE should generate the right nested column type. FE doesn't check the nesting depth and legality, like map\<array\<int\>, int\>.
2. `ParquetColumnReader` has removed the filtering of page index to support nested array type. It's too difficult to skip values in nested complex  types. Maybe we should support the filtering of page index and lazy read in later PR.
3. `ExternalFileScanNode` has a bug in creating default value expression.
4. Maybe it's slow to read repetition levels in a while loop. I'll optimize this in next PR.
5. Array column has temporary `SchemaElement` in its thrift definition, we have removed them and keep its parent in former implementation. The remaining parent should inherit the repetition and definition level of its child.

## Inherit repetition and definition level of child
The full schema for type `ARRAY<ARRAY<INT32>>` is:
```
 five levels for ARRAY<ARRAY<INT32>>
 income --- bag --- array_element --- bag --- array_element
  opt       rep          opt          rep         opt
 R=0,D=1  R=1,D=2       R=1,D=3     R=2,D=4      R=2,D=5
```
`bag` is the temporary `SchemaElement`. We should keep `income` and assign the R,D as its removed child `R=1,D=2`, so the final schema for type `ARRAY<ARRAY<INT32>>` is:
```
 three levels for ARRAY<ARRAY<INT32>>
 income --- array_element --- array_element
   rep           rep              opt
 R=1,D=2       R=2,D=4          R=2,D=5
```
## Record the definition level of parent
`repeated_parent_def_level` is the definition level of its parent `SchemaElement`, so the `R, D, P` in the final schema for type `ARRAY<ARRAY<INT32>>` is:
```
 three levels for ARRAY<ARRAY<INT32>>
   income ------ array_element ----- array_element
    rep              rep                 opt
 R=1,D=2,P=0      R=2,D=4,P=2         R=2,D=5,P=4
```
When definition level is less than `repeated_parent_def_level`, it means that level will affect its parent. When repetition level is greater than max_rep_level, it means that level affects its descendants.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by AshinGau** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.05 seconds
 stream load tsv:          460 seconds loaded 74807831229 Bytes, about 155 MB/s
 stream load json:         36 seconds loaded 2358488459 Bytes, about 62 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          27 seconds loaded 861443392 Bytes, about 30 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230222030022_clickbench_pr_101441.html

**Comment by AshinGau** [comment]
run buildall

**Comment by AshinGau** [comment]
run p0

**Comment by AshinGau** [comment]
run compile

File: fe/fe-core/src/main/java/org/apache/doris/catalog/HiveMetaStoreClientHelper.java [hunk 1]
```diff
@@ -705,6 +720,36 @@ public static Type hiveTypeToDorisType(String hiveType) {
                 return ArrayType.create(innerType, true);
             }
         }
+        if (lowerCaseType.startsWith("map")) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/HiveMetaStoreClientHelper.java:723]: better give an example in code comment

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**